bls/signature.go

```diff
@@ -41,15 +42,30 @@ func AggregateSignatures(signatures []*Signature) (*Signature, error) {
 	return &Signature{flowSignature: aggregateFlowSignature}, nil
 }
 
-// VerifyAggregateSignature takes in a slice of bls.PublicKeys, a bls.Signature, and a payload and returns
+// VerifyAggregateSignatureSinglePayload takes in a slice of bls.PublicKeys, a bls.Signature, and a single payload and returns
 // true if every bls.PublicKey in the slice signed the payload. The input bls.Signature is the aggregate
 // signature of each of their respective bls.Signatures for that payload.
-func VerifyAggregateSignature(publicKeys []*PublicKey, signature *Signature, payloadBytes []byte) (bool, error) {
+func VerifyAggregateSignatureSinglePayload(publicKeys []*PublicKey, signature *Signature, payloadBytes []byte) (bool, error) {
 	var flowPublicKeys []flowCrypto.PublicKey
 	for _, publicKey := range publicKeys {
 		flowPublicKeys = append(flowPublicKeys, publicKey.flowPublicKey)
 	}
-	return flowCrypto.VerifyBLSSignatureOneMessage(flowPublicKeys, signature.flowSignature, payloadBytes, HashingAlgorithm)
+	return flowCrypto.VerifyBLSSignatureOneMessage(flowPublicKeys, signature.flowSignature, payloadBytes, hashingAlgorithm)
+}
+
+// VerifyAggregateSignatureMultiplePayloads takes in a slice of bls.PublicKeys, a bls.Signature, and a slice of payloads.
+// It returns true if each bls.PublicKey at index i has signed its respective payload at index i in the payloads slice.
+// The input bls.Signature is the aggregate signature of each public key's partial bls.Signatures for its respective payload.
+func VerifyAggregateSignatureMultiplePayloads(publicKeys []*PublicKey, signature *Signature, payloadsBytes [][]byte) (bool, error) {
+	var flowPublicKeys []flowCrypto.PublicKey
+	for _, publicKey := range publicKeys {
+		flowPublicKeys = append(flowPublicKeys, publicKey.flowPublicKey)
+	}
```

Comment:
Starting to see this repeat. I think it's okay to have it duplicated for now, but if we continue to convert slices of PublicKey to slices of flowCrypto.PublicKey, we should write a little helper function.

Reply:
Good diea. Seems appropriate to create the helper function now.